src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp

```diff
@@ -70,33 +70,77 @@ namespace AirflowNetwork {
         Real64 viscosity{AIRDYNAMICVISCOSITY(20.0)};
     };
 
+    // Forward declaration
+    struct AirflowElement;
+
     struct Solver
     {
+        Solver() : PB(0.0)
+        {}
+
+        void allocate();
+        void initialize();
+        void setsky();
+        void airmov();
+        void solvzp(int &ITER);  // number of iterations
+        void filjac(int const NNZE,  // number of nonzero entries in the "AU" array.
+                    bool const LFLAG // if = 1, use laminar relationship (initialization).
+        );
+
+        void clear()
+        {
+            PB = 0.0;
+            //LIST = 0;
+            elements.clear();
+            compnum.clear();
+            properties.clear();
+            AFECTL.deallocate();
+            AFLOW2.deallocate();
+            AFLOW.deallocate();
+            PS.deallocate();
+            PW.deallocate();
+            SUMAF.deallocate();
+            PZ.deallocate();
+            ID.deallocate();
+            IK.deallocate();
+            AD.deallocate();
+            AU.deallocate();
+
+#ifdef SKYLINE_MATRIX_REMOVE_ZERO_COLUMNS
+            newIK.deallocate();
+            newAU.deallocate();
+#endif
+            SUMF.deallocate();
+        }
+
+        std::unordered_map<std::string, AirflowElement *> elements;
+        std::unordered_map<std::string, int> compnum; // Stopgap until all the introspection is dealt with
+
         std::vector<AirProperties> properties;
 
-        int NetworkNumOfLinks;
-        int NetworkNumOfNodes;
+        //int NetworkNumOfLinks;
+        //int NetworkNumOfNodes;
 
-        int const NrInt; // Number of intervals for a large opening
+        //int const NrInt; // Number of intervals for a large opening
```

Comment:
We should probably get rid of this.

Reply:
I'm leaving that in as a reminder that the global probably needs to go here. I've not had a great deal of luck messing around with detailed openings without causing diffs. So I'm avoiding it right now.